### PR TITLE
test: move ipcRenderer spec to main runner

### PR DESCRIPTION
**Pull request by nornagon**
#### Description of Change
Also simplify & robustify some tests. One of the tests in the suite was actually testing remote module stuff, so I moved it to the remote spec suite.

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] tests are [changed or added](https://github.com/electron/electron/blob/master/docs/development/testing.md)
- [ ] relevant documentation is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: none

### CODE REVIEW

File: spec-main/api-ipc-renderer-spec.ts [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+      }
+      w.webContents.executeJavaScript(`{
+        const { ipcRenderer } = require('electron')
+        ipcRenderer.send('message', ${JSON.stringify(obj)})
+      }`)
+      const [, received] = await emittedOnce(ipcMain, 'message')
+      expect(received).to.deep.equal(obj)
+    })
+
+    it('can send instances of Date as ISO strings', async () => {
```
Comment (jkleinsc) [inline spec-main/api-ipc-renderer-spec.ts]: Isn't it sending a Date object not an ISO string?

Comment (nornagon) [reply spec-main/api-ipc-renderer-spec.ts]: That's just how our IPC works 🤷‍♂ see https://github.com/electron/electron/pull/20214 for the PR that will make this send a real Date :)

File: spec-main/api-ipc-renderer-spec.ts [hunk 2]
```diff
@@ -0,0 +1,198 @@
+import { expect } from 'chai'
+//import * as http from 'http'
```
[the remark this answers is not in the document]
Comment (nornagon) [reply spec-main/api-ipc-renderer-spec.ts]: whoops, will delete :)
```suggestion
```

File: spec-main/api-ipc-renderer-spec.ts [hunk 3]
```diff
@@ -0,0 +1,197 @@
+import { expect } from 'chai'
+import * as path from 'path'
+import { ipcMain, /*webContents,*/ BrowserWindow, WebContents, WebPreferences, webContents } from 'electron'
```
Comment (nornagon) [inline spec-main/api-ipc-renderer-spec.ts]:
```suggestion
import { ipcMain, BrowserWindow, WebContents, WebPreferences, webContents } from 'electron'
```

**Pull request merged**